sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -83,52 +102,23 @@ export class CryptographyClient {
     plaintext: Uint8Array,
     options: EncryptOptions = {}
   ): Promise<EncryptResult> {
+    const localCryptographyClient = await this.getLocalCryptographyClient();
     const requestOptions = operationOptionsToRequestOptionsBase(options);
     const span = this.createSpan("encrypt", requestOptions);
 
-    if (isNode) {
-      await this.fetchFullKeyIfPossible();
-
-      if (typeof this.key !== "string") {
-        switch (algorithm) {
-          case "RSA1_5": {
-            if (this.key.kty !== "RSA") {
-              span.end();
-              throw new Error("Key type does not match algorithm");
-            }
-
-            if (this.key.keyOps && !this.key.keyOps.includes("encrypt")) {
-              span.end();
-              throw new Error("Key does not support the encrypt operation");
-            }
-
-            const keyPEM = convertJWKtoPEM(this.key);
-
-            const padded: any = { key: keyPEM, padding: constants.RSA_PKCS1_PADDING };
-            const encrypted = publicEncrypt(padded, Buffer.from(plaintext));
-            return { result: encrypted, algorithm, keyID: this.key.kid };
-          }
-          case "RSA-OAEP": {
-            if (this.key.kty !== "RSA") {
-              span.end();
-              throw new Error("Key type does not match algorithm");
-            }
-
-            if (this.key.keyOps && !this.key.keyOps.includes("encrypt")) {
-              span.end();
-              throw new Error("Key does not support the encrypt operation");
-            }
-
-            const keyPEM = convertJWKtoPEM(this.key);
-
-            const encrypted = publicEncrypt(keyPEM, Buffer.from(plaintext));
-            return { result: encrypted, algorithm, keyID: this.key.kid };
-          }
+    if (localCryptographyClient) {
+      try {
+        return localCryptographyClient.encrypt(algorithm as LocalSupportedAlgorithmName, plaintext);
+      } catch (e) {
+        if (!(e instanceof LocalCryptographyUnsupportedError)) {
```

Comment:
Thank you, Jeff. What is the concern here? that it won't work as expected? I can test that.

Reply:
Here are two examples of existing places in our codebase using this approach:

In `identity/src/credentials/interactiveBrowserCredential.browser.ts`:

https://github.com/Azure/azure-sdk-for-js/blob/1b7428dffc5568da7f1e886f48857c33aad2a411/sdk/identity/identity/src/credentials/interactiveBrowserCredential.browser.ts#L89-L102

In `core-amqp/src/errors.ts`:

https://github.com/Azure/azure-sdk-for-js/blob/ca4cc53a60219ea4c0fa8487f3e8e6b6991f0342/sdk/core/core-amqp/src/errors.ts#L635-L637

I do see one example of code using both instanceof and checking the error name, here in `servicebus/service-bus/src/core/messageSender.ts`:

https://github.com/Azure/azure-sdk-for-js/blob/7a0eaebc0e54ae8a6873391c44440d8dcd12e017/sdk/servicebus/service-bus/src/core/messageSender.ts#L607

Should I make a PR fixing the ones that do not check for the error name too?

Reply:
I'm curious what @bterlson thinks about it (when he gets back.)

The general reason you don't want to use `instanceof` at least for custom error types is that there are cases in JS where the check will fail unexpectedly (due to the lack of a global type registry.) For example, if two versions of one lib get included, and the error is created by one and received by the other, or if an error is create inside a different script context (e.g. iframe.)

It's not super common, but relying on instanceof for non-primitives has caused enough problems in the past.

Reply:
Checking for name AND instanceof is pointless, `name` should usually be enough. The issue is that there are cases where an instanceof check will fail unexpectedly, for example when there are side-by-side versions of the same library either because of the dependency graph or because you have multiple realms.

If you think you're getting some additional safety using instanceof, then you can use it. Otherwise just checking `name` should be sufficient most of the time.

Reply:
sounds good! thank you!